WooCommerce/WooCommerceTests/ViewRelated/Dashboard/Onboarding/StoreOnboardingViewModelTests.swift

```diff
@@ -1,64 +1,136 @@
 import XCTest
+import Yosemite
 @testable import WooCommerce
 
 final class StoreOnboardingViewModelTests: XCTestCase {
-    func test_numberOfTasksCompleted_returns_correct_count() {
+    private var stores: MockStoresManager!
+
+    override func setUp() {
+        super.setUp()
+        stores = MockStoresManager(sessionManager: SessionManager.makeForTesting())
+    }
+
+    override func tearDown() {
+        stores = nil
+        super.tearDown()
+    }
+
+    // MARK: - `numberOfTasksCompleted``
+
+    func test_numberOfTasksCompleted_returns_correct_count() async {
         // Given
-        let taskViewModels: [StoreOnboardingViewModel.TaskViewModel] = [
-            .init(task: .addFirstProduct, isComplete: true, icon: .productImage),
-            .init(task: .launchStore, isComplete: true, icon: .launchStoreImage),
-            .init(task: .customizeDomains, isComplete: false, icon: .domainsImage),
-            .init(task: .payments, isComplete: false, icon: .currencyImage)
-        ]
+        mockLoadOnboardingTasks(result: .success([
+            .init(isComplete: true, type: .addFirstProduct),
+            .init(isComplete: true, type: .launchStore),
+            .init(isComplete: false, type: .customizeDomains),
+            .init(isComplete: false, type: .payments)
+        ]))
         let sut = StoreOnboardingViewModel(isExpanded: true,
-                                           taskViewModels: taskViewModels)
+                                           siteID: 0,
+                                           stores: stores)
+        // When
+        await sut.reloadTasks()
 
         // Then
         XCTAssertEqual(sut.numberOfTasksCompleted, 2)
     }
 
-    func test_tasksForDisplay_returns_only_incomplete_tasks() {
+    // MARK: - `tasksForDisplay``
+
+    func test_tasksForDisplay_returns_first_3_incomplete_tasks_when_isExpanded_is_false() async {
         // Given
-        let taskViewModels: [StoreOnboardingViewModel.TaskViewModel] = [
-            .init(task: .addFirstProduct, isComplete: true, icon: .productImage),
-            .init(task: .launchStore, isComplete: false, icon: .launchStoreImage),
-            .init(task: .customizeDomains, isComplete: true, icon: .domainsImage),
-            .init(task: .payments, isComplete: false, icon: .currencyImage)
-        ]
+        mockLoadOnboardingTasks(result: .success([
+            .init(isComplete: false, type: .addFirstProduct),
+            .init(isComplete: false, type: .launchStore),
+            .init(isComplete: false, type: .customizeDomains),
+            .init(isComplete: false, type: .payments)
+        ]))
+        let sut = StoreOnboardingViewModel(isExpanded: false,
+                                           siteID: 0,
+                                           stores: stores)
+        // When
+        await sut.reloadTasks()
+
+        // Then
+        XCTAssertEqual(sut.tasksForDisplay.count, 3)
```

Comment:
nit: maybe we can also assert the task types in `tasksForDisplay` in case we're changing **which** 3 tasks to show in the future?